app/models/supervisor_run.rb

```diff
@@ -119,10 +120,23 @@ def monitoring_loop
     notify_error("Supervisor script is not running\nLast 100 lines of supervisor output:\n#{read_log}") unless check
   end
 
+  ##
+  # Stops supervisor run execution
+  def stop
+    return unless self.pid
+    return unless check
+    Process.kill('TERM', self.pid)
+    sleep 1
+    if check
+      Process.kill('INT', self.pid)
```

Comment:
Rozumiem, że INT miało wymuszać zniszczenie procesu po 1 sekundzie - w takim razie zrób KILL, bo INT jest to Ctrl+C z konsoli i tak jak pisze Mateusz - nie gwarantuje zniszczenia.